### PR TITLE
Add omaha/smart_proxy_omaha debian build

### PR DESCRIPTION
**Pull request by lhw**
Direct replacement for #3210 (I just squashed the three commits)
This plugin already exists in the RPM repos, supplied by @timogoebel. It can be build for:

* [x] Nightly
* [x] 1.21
* [x] 1.20
* [x] 1.19
* [x] 1.18

I am not to sure about compat and debian versions as the ones used here are fairly old but that might be because of the internal build system.

### CODE REVIEW

**Comment by ekohl** [comment]
ok to test

File: plugins/ruby-foreman-omaha/debian/control [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Source: ruby-foreman-omaha
+Section: ruby
+Priority: extra
+Maintainer: Lennart Weller <lennart.weller@hansemerkur.de>
+Build-Depends: debhelper (>= 8.0.0), foreman (>= 1.18.0~rc1), git
```
Comment (ekohl) [inline plugins/ruby-foreman-omaha/debian/control]: This needs `foreman-sqlite3` as well. Possibly foreman-assets

Comment (lhw) [reply plugins/ruby-foreman-omaha/debian/control]: Yeah sorry about the delay here. I'll get back on it in a bit.

File: plugins/ruby-foreman-omaha/debian/control [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Source: ruby-foreman-omaha
+Section: ruby
+Priority: extra
+Maintainer: Lennart Weller <lennart.weller@hansemerkur.de>
+Build-Depends: debhelper,
+               foreman (>= 1.18.0~rc1),
+               foreman-assets (>= 1.18.0~rc1),
+               foreman-sqlite3 (1.18.0~rc1)
```
Comment (mmoll) [inline plugins/ruby-foreman-omaha/debian/control]: needs `>= ` here as well

**Comment by lhw** [comment]
Alright. Finally. I don't know why the build server can't find the newer gem v3.0.0. But for now it works with v2.0.0. @ekohl maybe you know more about this?

File: plugins/ruby-foreman-omaha/debian/gem.list [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# list of gem urls to download via Jenkins, space separated
+GEMS="https://rubygems.org/downloads/foreman_omaha-2.0.0.gem"
```
Comment (ekohl) [inline plugins/ruby-foreman-omaha/debian/gem.list]: You forgot to bump this to 3.0.0 so it failed

Comment (lhw) [reply plugins/ruby-foreman-omaha/debian/gem.list]: Alright. Will fix that. So many different places to put stuff. Probably just lost sight of them all.

**Comment by lhw** [comment]
That should be all then.

File: plugins/ruby-foreman-omaha/debian/changelog [hunk 4]
```diff
@@ -0,0 +1,5 @@
+ruby-foreman-omaha (2.0.0-1) stable; urgency=medium
```
Comment (ekohl) [inline plugins/ruby-foreman-omaha/debian/changelog]: This should also be 3.0.0 I think

**Comment by mmoll** [comment]
I fixed up a fews thing, @lhw could you confirm that the latest DEBs at http://stagingdeb.theforeman.org/pool/plugins/lhw/r/ (20190129) work for you in a DEB production setup?

**Comment by ekohl** [comment]
Ping @lhw / @mmoll

**Comment by ekohl** [comment]
Do you have any logs? Can you curl `http://localhost:8080/features` and share that?

**Comment by mmoll** [comment]
[test packaging]

**Comment by lhw** [comment]
We had setup another host with the package from this PR and noticed a few things:
 - There is absolutely no error concerning omaha in the installer log
 - The UI elements (Hosts/Omaha Hosts and Configure/Omaha Groups) exists and snippets from the Omaha plugin also work in our provisioning templates
 - The smart proxy still isnt shown the features list

**Comment by mmoll** [comment]
@lhw please provide the logs that were requested long time ago.

**Comment by lhw** [comment]
https://gist.github.com/lhw/a10907579a3aec539f7829b13ccf10d4 Well here you go. I had to clean them up a bit. Note that the Omaha plugin itself works. It just doesn't show up in the list of smart proxies.

**Comment by mmoll** [comment]
The logs look good and in my local setup I also get "omaha" in the proxy while seeing nothing in the respective UI panel. So I'm merging this now to give this potential wider exposure...

**Comment by mmoll** [comment]
merged, danke @lhw!

**Comment by lhw** [comment]
> The logs look good and in my local setup I also get "omaha" in the proxy while seeing nothing in the respective UI panel. So I'm merging this now to give this potential wider exposure...

Yeah its really weird. It works fine but just won't show in the lists. So we had assumed that the install failed at first until I accidently used the template functions and it just worked.

**Comment by timogoebel** [comment]
That's what I get in our deployment:

image

**Comment by lhw** [comment]
Did you install with 1.21? Maybe whatever causes this issue is already fixed with 1.21. Our install is 1.20.2 as there were just too many api changes and I hadn't had the time yet.

**Comment by timogoebel** [comment]
This is a screenshot from a 1.20 install.

**Pull request merged**